src/vsr/replica.zig

```diff
@@ -8401,14 +8401,6 @@ pub fn ReplicaType(
                 return;
             }
 
-            if (candidate.view > self.view_durable()) {
-                // For ignoring, it is correct to check only view and not view_durable. This can't
-                // lead to a situation where we crash and restart with an older view and a newer
-                // sync target, because superblock updates are serialized.
-                assert(self.view > self.view_durable());
-                assert(self.view_durable_updating());
```

Comment:
Apparently, it can be the case that view > view_durable, but we are not updating the view. 

For example, we we concurrently bump the view twice, the second update might be lost:

https://github.com/tigerbeetle/tigerbeetle/commit/23a2aaeddd7639495b2de32e110789edfe06d35c#diff-e95e4dc6c1aae349e798cfeed22ad01f830cce0e654e6b4b765b3e7b0c677b6aR4969

Reply:
Hmm I'm confused... the second update is lost b/c we were are status=recovering_head?

Reply:
(Referring to https://github.com/tigerbeetle/tigerbeetle/blob/6defd9d8a73944934a9d01c774f170c7416bfb4a/src/vsr/replica.zig#L6901)

Reply:
So, if I apply the following diff to always chain-update view_durable:

```diff
diff --git a/src/vsr/replica.zig b/src/vsr/replica.zig
index d6bb4523..eae27635 100644
--- a/src/vsr/replica.zig
+++ b/src/vsr/replica.zig
@@ -6846,6 +6846,7 @@ pub fn ReplicaType(
             assert(self.log_view > self.log_view_durable() or self.view > self.view_durable());
             // The primary must only persist the SV headers after repairs are done.
             // Otherwise headers could be nacked, truncated, then restored after a crash.
+            assert(!((self.log_view == self.view and self.replica == self.primary_index(self.view) and self.status == .view_change)));
             assert(self.log_view < self.view or self.replica != self.primary_index(self.view) or
                 self.status == .normal or self.status == .recovering);
             assert(self.view_headers.array.count() > 0);
@@ -6908,7 +6909,7 @@ pub fn ReplicaType(
                 (self.replica != self.primary_index(self.view) or self.status == .normal);
             assert(!(update_dvc and update_sv));
 
-            if (update_dvc or update_sv) self.view_durable_update();
+            if (update) self.view_durable_update();
 
             // Reset SVC timeout in case the view-durable update took a long time.
             if (self.view_change_status_timeout.ticking) self.view_change_status_timeout.reset();
```

I get a crash on `15663507130681778245` on the newly added assert, which is a specialisation of the assert on the next line. 

So what happens is:

* something triggers a view change and replica goes and starts writing its new view to disk
* by the time update finishes, another view jump happned
* but for _this_ view, the replica is a primary
* and it would be incorrect for the primary to update view_durable _right now_, because that would also write view headers, but would-be primary doesn't necessary have all the headers required yet.
* that's why in this case the update is "lost", and is re-done explicitly in `transition_to_normal_from_view_change_status`.

Reply:
Raced! I am looking rather at the following line:

https://github.com/tigerbeetle/tigerbeetle/blob/6defd9d8a73944934a9d01c774f170c7416bfb4a/src/vsr/replica.zig#L6911C12-L6911C69